.gitlab/functional_test/regression_detector.yml

```diff
@@ -26,6 +26,13 @@ single-machine-performance-regression_detector:
   # See 'decision_record.md' for the determination of whether this job passes or fails.
   allow_failure: false
   script:
+    # `datadog-ci` relies on `DATADOG_API_KEY` so we get that here.
+    - DATADOG_API_KEY="$("$CI_PROJECT_DIR"/tools/ci/fetch_secret.sh "$AGENT_API_KEY_ORG2" token)" || exit $?; export DATADOG_API_KEY
+    # Start by tagging the failure mode as `unknown`. At each smp command we will catch any failure exit codes and update this tag properly.
+    # If we successfully get to the end of the script with no failures we will update the `smp_failure_mode` tag to `none` to convey this.
+    # Note, we cannot apply a tag for timeouts with this approach. In order to identify jobs failed by timing out you can filter by error type or duration.
+    # Note, in situations where the job times out the value of `smp_failure_mode` tag will be `unknown`.
```

Comment:
Would it make sense to start off the failure mode as `timeout` if that is the only case where we will not update it to a more specific failure mode? (or `none`)

Reply:
I think thats fine but one thing to note is that any other failures (besides the smp commands and bounds/opt goals/quality gates) would also fall under `timeout`. An example being [this python error](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/699309461#L123) which would also be classified as timeout then. We can leverage the error type in Ci Vis as seen [here](https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Ajob%20%40ci.job.name%3Asingle-machine-performance-regression_detector%20%40ci.status%3Aerror%20error.type%3Ajob_execution_timeout&agg_m=count&agg_m_source=base&agg_t=count&cipipeline_explorer_sort=timestamp%2Cdesc&colorBy=meta%5B%27ci.stage.name%27%5D&colorByAttr=meta%5B%27ci.stage.name%27%5D&cols=%40ci.status%2Ctimestamp%2C%40ci.pipeline.name%2C%40ci.stage.name%2C%40ci.job.name%2C%40duration%2C%40ci.pipeline.id%2C%40git.repository.name%2C%40git.branch&currentTab=trace&fromUser=false&graphType=flamegraph&index=cipipeline&spanViewType=metadata&start=1730404088306&end=1731700088306&paused=false) to filter on timed out jobs instead. I dont really love having two different types of signals as to how the jobs are failing but not sure there is a much better way to go about it that I can think of atm

Reply:
On a slightly related topic I think I want to move the update to `smp_failure_mode:none` before we run the quality gates script since if quality gates fail we will never update `smp_failure_mode` and we have the signal for quality gates in `smp_quality_gates:` anyways. Again not loving this cause then we have some logic to do to determine if a `smp_failure_mode:none` actually did "fail" due to a failed quality gate but the other choice means smp_quality_gates:failed will be paired with `smp_failure_mode:unknown` which doesnt make sense either. I am leaning towards the first pairing and treating `smp_failure_mode` as failures related to smp infra or other script failures, not regression/bound check/qg realted.